probes/webhooksWithoutTokenAuth/impl.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 OpenSSF Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint:stylecheck
+package webhooksWithoutTokenAuth
+
+import (
+	"embed"
+	"fmt"
+
+	"github.com/ossf/scorecard/v4/checker"
+	"github.com/ossf/scorecard/v4/finding"
+	"github.com/ossf/scorecard/v4/probes/internal/utils/uerror"
+)
+
+//go:embed *.yml
+var fs embed.FS
+
+const Probe = "webhooksWithoutTokenAuth"
+
+func Run(raw *checker.RawResults) ([]finding.Finding, string, error) {
+	if raw == nil {
+		return nil, "", fmt.Errorf("%w: raw", uerror.ErrNil)
+	}
+
+	r := raw.WebhookResults
+	totalWebhooks := len(r.Webhooks)
+	var webhooksWithoutSecret int
+
+	var findings []finding.Finding
+
+	if len(r.Webhooks) == 0 {
+		f, err := finding.NewWith(fs, Probe,
+			"Repository does not have webhooks.", nil,
+			finding.OutcomeNotApplicable)
+		if err != nil {
+			return nil, Probe, fmt.Errorf("create finding: %w", err)
+		}
+		findings = append(findings, *f)
+		return findings, Probe, nil
+	}
+
+	for _, hook := range r.Webhooks {
+		if !hook.UsesAuthSecret {
+			webhooksWithoutSecret++
+		}
+	}
+
+	if webhooksWithoutSecret != 0 {
+		whWoS, tW := webhooksWithoutSecret, totalWebhooks
+		msg := fmt.Sprintf("Repository has %d webhooks out of %d without token authorization.", whWoS, tW)
+		f, err := finding.NewWith(fs, Probe,
+			msg, nil, finding.OutcomeNegative)
+		if err != nil {
+			return nil, Probe, fmt.Errorf("create finding: %w", err)
+		}
+		f = f.WithValues(map[string]int{
+			"totalWebhooks":         totalWebhooks,
```

Comment:
see my comment in the yaml file. Should we return one finding per webhook? We want to be able to provide steps for remediation, and one of these steps _could_ be a link to the webhook configuration, based on the webhook name we know. 
Also, when creating bugs (in a company's internal bug tracking system or in GitHub scanning dashboard), users may want to track remediation for each webhook independently. The metadata is useful to have for this use case. It's possible that certain services support secrets, and others don't, so tracking each bug independently may be useful.

Curious what everyone thinks.

Reply:
Sounds like a good approach to me. @raghavkaul @spencerschrock WDYT?

Reply:
I think this is a scenario where 1 finding per webhook makes sense. As long as we make it possible to distinguish between them via some location/url/idenifier

Reply:
We will do one finding per webhook

Reply:
discussed with maintainers, 1 finding per webhook

Reply:
Done in https://github.com/ossf/scorecard/pull/3522/commits/cfb0caf8ef483fec819e50dd02eeb0191bd27a1b. The probe now returns one finding per webhook that does not have a secret configured.